lib/elixir/lib/module.ex

```diff
@@ -725,7 +738,10 @@ defmodule Module do
       end
 
   """
-  def defines?(module, tuple) when is_tuple(tuple) do
+  @spec defines?(module, function_arity) :: boolean
+  def defines?(module, {function_macro_name, arity} = tuple)
```

Comment:
I am worried that `function_name` doesn't take macros. That's why the difference.
We could name it `function_or_macro_name`